### PR TITLE
Require hosting deploy targets to be in the project's firebase.json

### PR DESCRIPTION
**Pull request by thomasmburke**
### Description

Bug fix for closed Github Issue [1232](https://github.com/firebase/firebase-tools/issues/1232). This is also tracked internally via b/163146235

When deploying using a hosting target not specified in the firebase.json hosting configuration file, currently users end up getting the following error:

`TypeError: Cannot read property 'deploys' of undefined`

This is because the code is left with an empty array [where the error is thrown](https://github.com/firebase/firebase-tools/blob/22ea3dc2b8eee19370f4ceeb5f54dbd6854022a9/src/deploy/index.js#L106), so to prevent this `TypeError` we can simply throw an error by checking that the hosting target is in the firebase.json before it gets there. 

This is much better behavior than what currently happens:
--> The CLI currently logs the "Deploy complete!" line just before throwing a type error.

### Scenarios Tested

- Tested with a hosting deploy target that is not specified in the firebase.json and also does not exist in the .firebaserc
- Tested with a hosting deploy target that is specified in the firebase.json and does not exist in the .firebaserc
- Tested with a hosting deploy target that is specified in the firebase.json and also exists in the .firebaserc
- Tested with a hosting deploy target that is not specified in the firebase.json and does exist in the .firebaserc
- Tested with multiple hosting deploy targets where one is valid (in both the firebase.json & .firebaserc) and one is invalid (missing from either the firebase.json or .firebaserc) (tested all permutations of the ways a target could be invalid from above)
- Tested with multiple hosting deploy targets where both are valid
- Tested with multiple hosting deploy targets where both are invalid
- Tested with no hosting deploy targets (`firebase deploy --only hosting` & `firebase deploy`)

### CODE REVIEW

**Comment by thomasmburke** [comment]
@googlebot I signed it!

**Review by samtstern — APPROVED** [review]
One nit, but approved.

File: src/hosting/normalizedHostingConfigs.ts [hunk 1]
```diff
@@ -19,10 +19,18 @@ function filterOnly(configs: HostingConfig[], onlyString: string): HostingConfig
...
   }
 
+  // Strip out Hosting deploy targets from onlyTarget
   onlyTargets = onlyTargets
     .filter((target) => target.startsWith("hosting:"))
     .map((target) => target.replace("hosting:", ""));
 
+  // Check to see that all the hosting deploy targets exist in the hosting config
+  onlyTargets.forEach((onlyTarget) => {
+    if (!configs.some((config) => config.target == onlyTarget)) {
```
[the remark this answers is not in the document]
Comment (thomasmburke) [reply src/hosting/normalizedHostingConfigs.ts]: Done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,2 +1,3 @@
-- Updates the Google Cloud Run proxy API calls to `v1` (from `v1alpha1`) (#2695).
-- Release RTDB emulator v4.6.0: Get wire protocol with optional query.
+- Updates the Google Cloud Run proxy API calls to `v1` (from `v1alpha1`) (#2695).
+- Release RTDB emulator v4.6.0: Get wire protocol with optional query.
+- Fixes `TypeError` that arises when trying to deploy with hosting targets that don't exist in the project's firebase.json (#1232).
```
Comment (bkendall) [inline CHANGELOG.md:3]:
```suggestion
- Fixes `TypeError` that arises when trying to deploy with Firebase Hosting targets that don't exist in the project's firebase.json (#1232).
```

**Review by bkendall — APPROVED** [review]

**Comment by samtstern** [comment]
@thomasmburke thank you!

**Pull request merged**